apps/api/src/modules/packs/packs.service.ts

```diff
@@ -749,7 +751,22 @@ export default class PacksService {
     const allPacks: PackByOwner[] = []
     for (const { id, claimedAt, templateId } of packs) {
       const template = templateLookup.get(templateId)
-      if (template && claimedAt) {
+      // For packs that require payment, we need to check if the user has paid
```

Comment:
This is to fix for a case where people where able to claim (transfer to their custodial) before their payment was marked as confirmed or paid.

Reply:
This logic actually needs to apply to auctions too I think, though I didn't look to see if it's the same

Reply:
I think you're right, and my guess is it needs to be exactly the same for auctions. However, I'm not sure if any of this is different for crypto and wire payments. @dallashuggins?

Reply:
I also made these changes in my PR to avoid some edge cases: https://github.com/deptagency/algomart/pull/155/files#diff-9163417dd8355447ea09b0ec98fb195305136b1506876ef2c02ddfbecde5a1ceL720-R733

Reply:
great, we can just address these specific changes in your PR